### PR TITLE
Purge all

### PR DESCRIPTION
**Pull request by Sue-Gallagher**
[DOCS] Removed or rewrote references to _all field.

### CODE REVIEW

**Comment by rjernst** [comment]
@Sue-Gallagher This PR looks like it is either not based on `master`, or based on a very old version. Can you bring it up to date or recreate it?

File: docs/reference/search/explain.asciidoc
Comment (debadair) [inline docs/reference/search/explain.asciidoc]: api -> API

This is somewhat misleading/overly simplistic. Explain calculates the relevance score for a particular document and shows what led to that score. It's not just whether or not the doc matches the query. The idea is to help you understand why a doc is being scored the way it is (and why it might or might not be surfaced in results the way you expect).

For context, see https://www.elastic.co/guide/en/elasticsearch/guide/current/relevance-intro.html#explain

I'd probably change this to something like:

> The explain API calculates the relevance score for a particular document and shows what led to that score. This can help you understand why a document is (or isn't) showing up in the search results for a particular query.

File: docs/reference/search/explain.asciidoc
Comment (debadair) [inline docs/reference/search/explain.asciidoc]: I find it misleading to refer to index and type as parameters, but have a section called "All Parameters" that doesn't mention them (because they aren't query string parameters, they're built into the URL). My preference would be to write around it, something like "You must run explain against a particular document. You cannot specify multiple indices, types, or documents." (I don't have the stack running to verify, but I'm pretty sure you can't specify multiple docs...)

File: docs/reference/search/explain.asciidoc
Comment (debadair) [inline docs/reference/search/explain.asciidoc]: Maybe...

> The `q` parameter provides a shortcut for specifying the query. The following request is equivalent to the full `query_string` query in the previous example:

File: docs/reference/search/explain.asciidoc
Comment (debadair) [inline docs/reference/search/explain.asciidoc]: Instead of "document explained", I'd probably say:

Set to `true` to retrieve the document `_source` along with the explanation. To retrieve a subset of the document `_source`, you can use `_source_include` and `_source_exclude`. For more information, see the <<get-source-filtering,Get API>>.

File: docs/reference/search/explain.asciidoc
Comment (debadair) [inline docs/reference/search/explain.asciidoc]: "Allows to"  is one of my least-favorite patterns in the docs. I'd opt for "Controls which stored fields are returned with the explanation." Or maybe just "The stored fields to return with the explanation."

File: docs/reference/search/explain.asciidoc
Comment (debadair) [inline docs/reference/search/explain.asciidoc]: "Controls which", or maybe just "The shard where you want to run the explain operation."

File: docs/reference/search/explain.asciidoc
Comment (debadair) [inline docs/reference/search/explain.asciidoc]: I'd test this out to see what it actually does. It's not clear to me what "the data of the request" is. Is it actually referring to the response?

File: docs/reference/search/explain.asciidoc
Comment (debadair) [inline docs/reference/search/explain.asciidoc]: Another construct I'm not fond of "Shoud xyz or not." I'd probably go with, "Analyze wildcard and prefix queries."

File: docs/reference/search/explain.asciidoc
Comment (debadair) [inline docs/reference/search/explain.asciidoc]: Similar to above. I'd probably go with, "Ignore format based failures such as providing text to a numeric field.

**Review by debadair — COMMENTED** [review]
Most of these have very little to do with the specific goal of this PR & are more to give you a sense for how (and how much) I think we should be editing things. They're things I'd probably fix while I was in there. But keep in mind that it can make it harder for reviewers to discern what the substantive changes are. In some cases, you might want to hold off on the editorial stuff until the more substantive changes have been reviewed.

File: docs/reference/search/explain.asciidoc
Comment (debadair) [inline docs/reference/search/explain.asciidoc]: Typo: SSet

**Review by debadair — APPROVED** [review]
One typo that needs fixed, other than that LGTM.

File: docs/reference/query-dsl/mlt-query.asciidoc
Comment (debadair) [inline docs/reference/query-dsl/mlt-query.asciidoc]: I think the extra line break introduced here is what's causing the doc test error.

**Comment by rjernst** [comment]
Please use a more meaningful commit message (when clicking the merge button, you can change this in the text box GitHub provides). Something like `Docs: Remove remaining references to _all field`.

**Comment by rjernst** [comment]
Also, elasticsearch-ci needs to pass before merging. The current error is because the yaml test cannot be generated because of encoding in a docs file:

```
22:54:25 ERROR   0.01s | DocsClientYamlTestSuiteIT.initializationError <<< FAILURES!
22:54:25    > Throwable #1: java.io.IOException: Error parsing reference/query-dsl/mlt-query
22:54:25    > 	at org.elasticsearch.test.rest.yaml.section.ClientYamlTestSuite.parse(ClientYamlTestSuite.java:71)
22:54:25    > 	at org.elasticsearch.test.rest.yaml.ESClientYamlSuiteTestCase.createParameters(ESClientYamlSuiteTestCase.java:154)
22:54:25    > 	at org.elasticsearch.smoketest.DocsClientYamlTestSuiteIT.parameters(DocsClientYamlTestSuiteIT.java:44)
22:54:25    > 	at java.lang.reflect.Constructor.newInstance(Constructor.java:423)
22:54:25    > Caused by: ParsingException[Error parsing test named [line_69]]; nested: IOException[Error parsing section starting at [69:5]]; nested: JsonParseException[Illegal unquoted character ((CTRL-CHAR, code 10)): has to be escaped using backslash to be included in string value
22:54:25    >  at [Source: java.io.StringReader@45d2ade3; line: 14, column: 78]];
22:54:25    > 	at org.elasticsearch.test.rest.yaml.section.ClientYamlTestSection.parse(ClientYamlTestSection.java:50)
22:54:25    > 	at org.elasticsearch.test.rest.yaml.section.ClientYamlTestSuite.parse(ClientYamlTestSuite.java:95)
22:54:25    > 	at org.elasticsearch.test.rest.yaml.section.ClientYamlTestSuite.parse(ClientYamlTestSuite.java:69)
22:54:25    > 	... 22 more
22:54:25    > Caused by: java.io.IOException: Error parsing section starting at [69:5]
22:54:25    > 	at org.elasticsearch.test.rest.yaml.section.ExecutableSection.parse(ExecutableSection.java:58)
22:54:25    > 	at org.elasticsearch.test.rest.yaml.section.ClientYamlTestSection.parse(ClientYamlTestSection.java:41)
22:54:25    > 	... 24 more
22:54:25    > Caused by: com.fasterxml.jackson.core.JsonParseException: Illegal unquoted character ((CTRL-CHAR, code 10)): has to be escaped using backslash to be included in string value
22:54:25    >  at [Source: java.io.StringReader@45d2ade3; line: 14, column: 78]
```

**Review by jimczi — CHANGES_REQUESTED** [review]
I see only one change related to `_all` in the `copy_to` section and this change adds a reference to it.
Am I missing something ? I guess this pr is more a cleanup with various changes that are not related to each other so can you change the title and make it clear that the changes are just cosmetics ? There are also some editorial changes that should be reviewed more carefully like the change in `aliases.asciidoc`. It might be better to open a separate pr for them.

**Pull request closed without merging**
